src/main/java/seedu/address/model/person/Availability.java

```
@@ -2,55 +2,63 @@
+import java.time.DayOfWeek;
+import java.time.format.TextStyle;
 import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
 
 /**
  * Represents a Person's availability in the address book.
  * Guarantees: immutable; is always valid
  */
 public class Availability {
 
-    public static final String[] DAYS = {"MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"};
-
     public static final String MESSAGE_CONSTRAINTS =
-            "Availability should be given as the abbreviated names of days(mon tue wed thu fri sat sun)";
+            "Availability should be given as a list of numbers separated by a space each, "
+            + "where 1 represents Monday, 2 represents Tuesday... and 7 represents Sunday";
 
-    public final String values;
+    public final List<DayOfWeek> values;
 
     /**
      * Constructs an {@code Availability}.
      *
      * @param availability A valid availability string.
      */
-    public Availability(String availability) {
+    public Availability(List<DayOfWeek> availability) {
         requireNonNull(availability);
         values = availability;
     }
 
     /**
-     * Returns true if a given availability is valid.
+     * Returns true if a given availability list is valid.
      */
     public static boolean isValidAvailability(List<String> test) {
-        if (test.size() == 1) {
-            return true;
-        }
-        List<String> days = Arrays.asList(DAYS);
-        for (String day : test) {
-            if (!days.contains(day)) {
-                return false;
+        try {
+            if (test.get(0).isEmpty()) { // empty but valid
+                return true;
+            }
+            for (String s : test) {
+                int dayNumber = Integer.parseInt(s);
+
+                if (dayNumber < 1 || dayNumber > 7) {
+                    return false;
+                }
             }
+        } catch (NumberFormatException e) { // if any of the string is not a number, it is invalid
```

Perhaps this should throw an error instead since try catch is not supposed to be used for handling logic?

---

hmm, throwing an error seems slightly odd since this is a boolean function and is not caught anywhere. Maybe i will try to validate with regex first so we dont have to throw an exception for ParseInt instead?